index.d.ts

```diff
@@ -0,0 +1,33 @@
+declare class MapExpander {
+  expandMap(currentMap: Object, pathString: string, mapValue: Object): Object;
+  getMapValue(currentMap: Object, pathString: string): Object;
+}
+
+declare class Response {
+  constructor(body: Object, statusCode: number, headers: Object): void;
+}
+
+declare class ApiFactory {
+  constructor(options: object, overrideLogger: object);
+
+
+  setAuthorizer(authorizerFunc: any): void;
+  SetAuthorizer(authorizerFunc: any): void;
+
+  onEvent(onEventFunc: any): void;
+
+  onSchedule(onScheduleFunc: any): void;
+
+  head(route: string, p0: Function, p1: Function): void;
```

Comment:
These will actually cause a problem I believe.  The `p0` value is optional, which means it can either be an Object or a Function, and when it is a Function, then `p1` is null.  I bet the right way to handle that in typescript is to create two copies of each method, where the second one is an overload. i.e.
* `get(route: string, handlerFunction: Function): void;`
* `get(route: string, options: object, handlerFunction: Function): void;`